### DISCUSSION TITLE
How to get started and debug ('Ledvance' object has no attribute 'get')

### DISCUSSION THREAD

**Discussion by gnbl** [kardia-as/zigpy-zboss#55] [Q&A]
Hi,

I've just flashed https://github.com/kardia-as/nrf-zboss-ncp/blob/main/nRF52840_dongle/v2.6.0_ncp_zboss_max_mem_nrf52840_dongle.hex to a new nRF52840-Dongle using 'nrfconnect' and installed https://github.com/kardia-as/zigpy-zboss and zigpy-0.65.4 (etc.) for Python 3.10.12 on LinuxMint.

The dongle shows up as

```
[ 1035.984219] usb 3-1.2: new full-speed USB device number 6 using xhci_hcd
[ 1036.118535] usb 3-1.2: New USB device found, idVendor=2fe3, idProduct=0001, bcdDevice= 3.05
[ 1036.118540] usb 3-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1036.118544] usb 3-1.2: Product: Zigbee NCP
[ 1036.118546] usb 3-1.2: Manufacturer: ZEPHYR
[ 1036.118549] usb 3-1.2: SerialNumber: C71Axxx
[ 1036.184623] cdc_acm 3-1.2:1.0: ttyACM0: USB ACM device
```

and `Bus 003 Device 008: ID 2fe3:0001 NordicSemiconductor Zigbee NCP`.

I've also installed `zigpy_cli-1.0.5`.

When I try `zigpy -vv radio zboss /dev/ttyACM0 info` it fails with 

```
...
  File "/home/user/.local/lib/python3.10/site-packages/zigpy/config/validators.py", line 124, in cv_ota_provider
    provider_type = obj.get(zigpy.config.CONF_OTA_PROVIDER_TYPE)
AttributeError: 'Ledvance' object has no attribute 'get'
```

Any hints on how to verify that the firmware works and how to narrow down the error would be appreciated.

**Comment by puddly** [comment]
This library probably needs a PR similar to this: <https://github.com/zigpy/zigpy-znp/pull/250/files>

**Comment by gnbl** [comment]
Thanks, applying that to `https://github.com/kardia-as/zigpy-zboss/blob/main/zigpy_zboss/zigbee/application.py#L44` gets me a bit further, but ends with apparently unhandled exception:

```
  File "/home/user/.local/lib/python3.10/site-packages/zigpy_zboss/zigbee/application.py", line 302, in load_network_info
    raise zigpy.exceptions.NetworkNotFormed
zigpy.exceptions.NetworkNotFormed
```

I'll try to investigate further when I have time.
Not sure what is expected to happen with this command (I have a few Ikea devices, some paired to the older Ikea hub, some un-paired).

**Reply by tube0013** [reply]
That's expected you need to form a network before zigpy can read its info

**Reply by gnbl** [reply]
Thanks, I'll look into that.
